tests/ballerina-integration-test-utils/src/main/java/org/ballerinalang/test/context/ServerInstance.java

```diff
@@ -276,6 +273,83 @@ public void runMain(String[] args, String[] envVariables, String command) throws
         }
     }
 
+    /**
+     * Run command with client options.
+     *
+     * @param args         client arguments
+     * @param options      options
+     * @param envVariables environment variables
+     * @param command      command name
+     * @param dir          working directory name
+     * @throws BallerinaTestException
+     */
+    public void runMainWithClientOptions(String[] args, String[] options, String[] envVariables, String command,
+                                         String dir) throws BallerinaTestException {
+        initialize();
+        File commandDir = new File(dir);
+        Process process;
+
+        try {
+            process = executeProcess(args, envVariables, command, commandDir);
+            // Wait until the options are prompted
+            Thread.sleep(3000);
+            writeClientOptionsToProcess(options, process);
+            deleteWorkDir();
+        } catch (IOException e) {
+            throw new BallerinaTestException("Error executing ballerina", e);
+        } catch (InterruptedException ignore) {
+        }
+    }
+
+    /**
+     * Execute process.
+     *
+     * @param args         client arguments
+     * @param envVariables environment variables
+     * @param command      command name
+     * @param commandDir   working directory
+     * @return process executed
+     * @throws IOException
+     */
+    private Process executeProcess(String[] args, String[] envVariables, String command, File commandDir)
+            throws IOException {
+        String scriptName = Constant.BALLERINA_SERVER_SCRIPT_NAME;
+        String[] cmdArray;
+        Process process;
+        if (Utils.getOSName().toLowerCase(Locale.ENGLISH).contains("windows")) {
+            cmdArray = new String[]{"cmd.exe", "/c", serverHome + File.separator + "bin" + File.separator + scriptName
+                    + ".bat", command};
+            String[] cmdArgs = Stream.concat(Arrays.stream(cmdArray), Arrays.stream(args))
+                                     .toArray(String[]::new);
+            process = Runtime.getRuntime().exec(cmdArgs, envVariables, commandDir);
+
+        } else {
+            cmdArray = new String[]{"bash", serverHome + File.separator + "bin/" + scriptName, command};
+            String[] cmdArgs = Stream.concat(Arrays.stream(cmdArray), Arrays.stream(args))
+                                     .toArray(String[]::new);
+            process = Runtime.getRuntime().exec(cmdArgs, envVariables, commandDir);
```

Comment:
Can you check whether theres an advantage of using 
`Process process = new ProcessBuilder("processname").start();`
against
`Runtime.getRuntime().exec(cmdArgs, envVariables, commandDir`

Refer: https://stackoverflow.com/questions/3774432/starting-a-process-in-java

Reply:
I'll look into the advantages